File: Practices/time_of_day.py
# IB 7th period Time of Day - python

from datetime import datetime

# Get the current date and time
now = datetime.now()

# Access the hour attribute
current_hour = now.hour


def unknown_time():
    print("Good think I know,")
    if time < 12:
     print("Good Morning!")

    elif time >= 12 and time < 18:
     print("Good Afternoon!")

    elif time < 0 or time > 24:
        print("That is not a time.")

    else:
     print("Good Evening!")

def known_time():
    if time_input < 12:
     print("Good Morning!")

    elif time_input >= 12 and time_input < 18:
     print("Good Afternoon!")

    elif time_input < 0 or time_input > 24:
        print("That is not a time.")

    else:
     print("Good Evening!")


do_you_know = input("Do you know what time it is (Yes/No): ").strip().capitalize()

if do_you_know == "Yes":
   time_input = int(input("What is the hour in military time: "))
   known_time()

else:
  time = current_hour
  unknown_time()
  

File: C/Practices/time_of_day.c
// IB 7th - Time of Day
#include <stdio.h>
#include <time.h>

int main() {

    char answer;
    int hour;

    printf("Do you know what time it is? (Y/N): ");
    scanf(" %c", &answer);

    if (answer == 'Y' || answer == 'y' || answer == 'yes' || answer == 'YES' || answer == 'Yes') {

        printf("What is the hour in military time (0-23): ");
        scanf("%d", &hour);

    } 
    else {

        printf("Good thing I know...\n");

        time_t now = time(NULL);
        struct tm *local = localtime(&now);

        hour = local->tm_hour;

        printf("The current hour is: %d\n", hour);
    }

    if (hour < 0 || hour > 23) {
        printf("That is not a valid time.\n");
    }
    else if (hour < 12) {
        printf("Good Morning!\n");
    }
    else if (hour < 18) {
        printf("Good Afternoon!\n");
    }
    else {
        printf("Good Evening!\n");
    }

    return 0;
}